### PR TITLE
Add AggregationCenter class which deals with grouping the jobs based on different criteria

### PR DESCRIPTION
**Pull request by andreeanica16**
1. AggregationCenter class has methods aggregateByUser (which groups by user) and aggregateByName (which groups by the name of the job).
2. AggregationCenterTest which checks aggregateByName (1 test that checks identifying the name right)

### CODE REVIEW

**Comment by andreeanica16** [comment]
@iht @juliocc @tblanshard

**Review by juliocc — APPROVED** [review]
LGTM

File: src/main/java/com/google/sps/AggregationCenter.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            groupedJobs.put(key, list);
+        }
+      }
+    }
+
+    return groupedJobs;
+  }
+
+  public Map<String, List<JobJSON>> aggregateByUser(List<JobJSON> jobs) {
+      return agreggateByGroupIndex(jobs, 2);
```
Comment (tblanshard) [inline src/main/java/com/google/sps/AggregationCenter.java:53]: Not a change, but I was just curious about how your index variable works - I had a quick read about the Matcher.group() function but was still a little confused having never used it before, and would love to learn more about it! :)

**Review by tblanshard — APPROVED** [review]
LGTM :)

**Pull request merged**